### PR TITLE
More tidying prior to Models in Browse Data

### PR DESCRIPTION
**Pull request by rafpaf**
The PR makes some e2e tests and associated components more semantic

### CODE REVIEW

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js [hunk 1]
```diff
@@ -135,15 +137,14 @@ describeEE("scenarios > embedding > full app", () => {
...
         qs: { side_nav: false, logo: false },
       });
-      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-      cy.findByText("Our data").should("be.visible");
-      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-      cy.findByText("Our analytics").should("not.exist");
+      cy.findByRole("heading", { name: /Our data/ }).should("be.visible");
+      cy.findByRole("treeitem", { name: /Our data/ }).should("not.exist");
+      cy.findByRole("treeitem", { name: "Our analytics" }).should("not.exist");
       appBar().should("not.exist");
```
Comment (rafpaf) [inline e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:148]: I'm not totally sure I've understood what this test is about

File: e2e/test/scenarios/onboarding/auth/signin.cy.spec.js [hunk 2]
```diff
@@ -30,40 +30,36 @@ describe("scenarios > auth > signin", () => {
...
     cy.findByLabelText("Email address").type(admin.email);
     cy.findByLabelText("Password").type("INVALID" + admin.password);
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.findByText("Sign in").click();
-    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-    cy.contains("did not match stored password");
+    cy.button("Sign in").click();
+    cy.findByRole("alert")
+      .filter(':contains("did not match stored password")')
+      .should("be.visible");
```
Comment (rafpaf) [inline e2e/test/scenarios/onboarding/auth/signin.cy.spec.js:36]: `cy.findByRole("alert", {name: ...` does not work; this longer formula is needed.

**Review by iethree — CHANGES_REQUESTED** [review]
looks great, except we don't want to keep adding more chainers off of `cy.`

File: e2e/support/commands/ui/button.ts [hunk 3]
```diff
@@ -34,4 +46,21 @@ Cypress.Commands.add(
...
+      name: headingName,
+      timeout,
+    };
+
+    return subject
+      ? cy.wrap(subject).findByRole("heading", config)
+      : cy.findByRole("heading", config);
+  },
+);
+
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply e2e/support/commands/ui/button.ts:65]: Yup, what Ryan said.
If anything, we want to reduce the amount of custom commands.

`findByRole` is already a custom command - it comes from [Cypress Testing Library](https://testing-library.com/docs/cypress-testing-library/intro/).

Can we please use it directly as is?

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:148]: IIRC, @WiNloSt originally wrote these tests.
I think the test is doing what it says on the tin - if we hide the side navigation **and** hide the logo, we want to make sure the top bar is not rendered.

There is a test below that explicitly makes sure nav bar is hidden via param.

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js (hunk 1 above)
Comment (rafpaf) [reply e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:159]: Thank you for helping me understand. I've reverted that change

File: e2e/support/commands/ui/button.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (rafpaf) [reply e2e/support/commands/ui/button.ts]: Thanks both. I've removed that extra helper function.

**Review by iethree — APPROVED** [review]
🚀

File: e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js (hunk 1 above)
Comment (WiNloSt) [reply e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js:159]: @nemanjaglumac got it exactly right. That test was added during the implementation of hiding Metabase logo.

For context if you're interested:
- [PR](https://github.com/metabase/metabase/pull/27391)
- [Issue](https://github.com/metabase/metabase/issues/27293)

**Review by nemanjaglumac — APPROVED** [review]
Thanks for improving a11y and for addressing the reviews.

**Pull request merged**